packages/rum-core/src/domain/configuration.ts

```diff
@@ -144,6 +146,109 @@ export function validateAndBuildRumConfiguration(
   )
 }
 
+/**
+ * Handles configureTracingUrls and processes legacy allowedTracingOrigins
+ */
+function handleTracingParameters(initConfiguration: RumInitConfiguration): ConfigureTracingOption[] | undefined {
+  // Advise about parameters precedence.
+  if (initConfiguration.configureTracingUrls !== undefined && initConfiguration.allowedTracingOrigins !== undefined) {
+    display.warn(
+      'Both configureTracingUrls and allowedTracingOrigins (deprecated) have been defined. The parameter configureTracingUrls will override allowedTracingOrigins.'
+    )
+  }
+
+  // Handle configureTracingUrls first
+  if (initConfiguration.configureTracingUrls !== undefined) {
+    if (!Array.isArray(initConfiguration.configureTracingUrls)) {
+      display.error('Configure Tracing URLs should be an array')
+      return
+    }
+    if (initConfiguration.configureTracingUrls.length !== 0 && initConfiguration.service === undefined) {
+      display.error('Service needs to be configured when tracing is enabled')
+      return
+    }
+
+    // Convert from MatchOption | TracingOption to TracingOption
+    const configureTracingUrls: ConfigureTracingOption[] = []
+    for (const option of initConfiguration.configureTracingUrls) {
+      configureTracingUrls.push(
+        isMatchOption(option) ? ({ match: option, headerTypes: ['dd'] } as ConfigureTracingOption) : option
+      )
+    }
+    return configureTracingUrls
+  }
+
+  // Handle conversion of allowedTracingOrigins to configureTracingUrls
+  if (initConfiguration.allowedTracingOrigins !== undefined) {
+    if (!Array.isArray(initConfiguration.allowedTracingOrigins)) {
+      display.error('Allowed Tracing Origins should be an array')
+      return
+    }
+    if (initConfiguration.allowedTracingOrigins.length !== 0 && initConfiguration.service === undefined) {
+      display.error('Service needs to be configured when tracing is enabled')
+      return
+    }
+
+    return initConfiguration.allowedTracingOrigins.reduce((configArray, item) => {
+      const option = convertLegacyMatchOptionToTracingOption(item)
+      if (option) {
+        configArray.push(option)
+      }
+      return configArray
+    }, [] as ConfigureTracingOption[])
+  }
+
+  return []
+}
+
+/**
+ * Converts parameters from the deprecated allowedTracingOrigins
+ * to configureTracingUrls. Handles the change from origin to full URLs.
+ */
+function convertLegacyMatchOptionToTracingOption(item: MatchOption) {
+  let match: MatchOption | undefined
+  if (typeof item === 'string') {
+    match = item
+  }
+  if (item instanceof RegExp) {
+    match = (url) => item.test(getOrigin(url))
+  }
+  if (typeof item === 'function') {
+    match = (url) => item(getOrigin(url))
+  }
+
+  if (match === undefined) {
+    display.warn('Allowed Tracing Origins parameters should be a string, RegExp or function. Ignoring parameter', item)
+    return undefined
+  }
+
+  return { match, headerTypes: ['dd'] } as ConfigureTracingOption
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏By defining the return type explicitly in the function signature, you could remove the need for a cast here. This would improve type safety because TS would check whether the return value matches the wanted type instead of forcing it to convert it.

Reply:
👌